mlos_bench/mlos_bench/tests/optimizers/mock_opt_test.py

```diff
@@ -71,6 +80,15 @@ def test_mock_optimizer(mock_opt: MockOptimizer, mock_configurations: list) -> N
     assert score == pytest.approx(66.66, 0.01)
 
 
+def test_mock_optimizer_no_defaults(mock_opt_no_defaults: MockOptimizer,
+                                    mock_configurations_no_defaults: list) -> None:
+    """
+    Make sure that mock optimizer produces consistent suggestions.
+    """
+    score = _optimize(mock_opt_no_defaults, mock_configurations_no_defaults)
+    assert score == pytest.approx(66.66, 0.01)
```

Comment:
Is it expected that these still come up with the same score?

Reply:
I guess my concern is that right now there isn't a good test that I've seen so far that `start_with_defaults` actually returns the defaults for the first iteration (but not when bulk registered).
Maybe we just need explicit tests for those cases.

Reply:
why, that's exactly the test - we have two instances of MockOptimizer, one for each value of `start_with_defaults`, and we check what values they suggest at the first iteration

Reply:
I missed the details in the `_optimize()` call (which you didn't change) already does an `assert` that the `suggest` value returned matches those in the listed mock_configs.  I thought we were only checking that the got some reasonable looking score when registering those.

But do we have one that tests the logic when bulk_registering is in effect?

Reply:
good idea! just added a test for it